src/config/mod.rs

```diff
@@ -0,0 +1,9 @@
+/// Configuration module. This module is used to parse configuration from
+/// command line arguments
+mod config;
+/// Inner configuration module. This module is used to store the parsed
+/// configuration and will be used to initialize the server
+mod inner;
+
+pub use config::Config;
+pub use inner::{InnerConfig, Role as NodeRole, StorageConfig, StorageParams, StorageS3Config};
```

Comment:
Why we rename `Role` as `NodeRole`? Does it conflict with other items?

Reply:
I will not rename it. But it seems this is not a good name either. I am looking for a better name.